nessie/system_test.go

```diff
@@ -75,22 +89,35 @@ func createRepository(ctx context.Context, t *testing.T, name string, repoStorag
 		StorageNamespace: repoStorage,
 	})
 	require.NoErrorf(t, err, "failed to create repository '%s', storage '%s'", name, repoStorage)
-	require.Equalf(t, http.StatusCreated, resp.StatusCode(),
-		"failed to create repository '%s', storage '%s': response %s", name, repoStorage, string(resp.Body))
+	require.NoErrorf(t, verifyResponse(resp.HTTPResponse, resp.Body),
+		"create repository '%s', storage '%s'", name, repoStorage)
 }
 
-func uploadFileRandomDataAndReport(ctx context.Context, repo, branch, objPath string) (checksum, content string, err error) {
+func uploadFileRandomDataAndReport(ctx context.Context, t *testing.T, repo, branch, objPath string, direct bool) (checksum, content string, err error) {
 	const contentLength = 16
 	objContent := randstr.Hex(contentLength)
 
-	resp, err := uploadContent(ctx, repo, branch, objPath, objContent)
-	if err != nil {
-		return "", "", err
-	}
-	if resp.StatusCode() != http.StatusCreated {
-		return "", "", fmt.Errorf("%w: %s (%d)", errUploadFailed, resp.Status(), resp.StatusCode())
+	if direct {
+		stats, err := helpers.ClientUpload(ctx, client, repo, branch, objPath, nil, strings.NewReader(objContent))
+		if errors.Is(err, helpers.ErrUnsupportedProtocol) {
+			t.Logf("direct upload not supported yet (%s), switch to use gateway", err)
+			directUploadSupported = false
+			return uploadFileRandomDataAndReport(ctx, t, repo, branch, objPath, false)
```

Comment:
This is an indirect way to understand if the current settings support direct upload or not.
Since we're trying to test the actual system, it might ignore real errors. For instance, s3 returns `ErrUnsupportedProtocol`. 

Any chance we can pass it as a test argument and expects all actions to succeed (or fail) according to the prior expectation, and not one inferred from the first call?